### ISSUE TITLE
Skip running App.Before function when --help is set on a sub-command

### ISSUE THREAD

**Issue by moorara** [urfave/cli#526]
Let's say I have an app called `cli` with a `Before` function:

```
app.Before = func(context *cli.Context) error { 
     // some initializations
}
```

If run `cli --help` or `cli --help subcmd`, the `Before` function won't be run and fine! But, if I run `cli subcmd --help` the `Before` will get executed while I'm still only interested in seeing the help for subcommand!

The `Before` function should actually be executed before any subcommand, but not when I want to see only help. Any workaround?

**Comment by jszwedko** [comment]
Hi @moorara,

I agree that this is a bit confusing. In your latter example (`cli subcmd --help`) the `--help` is actually processed and handled by the subcommand itself which causes the `App` to run the `Before` handler before executing. In the other example, the `App` itself is handling the `--help` flag which causes it to exit before starting to handle the subcommand. I'll have to think a bit more about what the expected behavior should be (I can see an argument for both ways), but in the meantime a, admittedly far from ideal, workaround would be to look at `context.Args()` and see if `--help` or `-h` is set anywhere in the string.

**Comment by lynncyrin** [comment]
Given that this is from last year, I think I'm comfortable closing it 🙂 feel free to re-open / open a new issue / comment in support if there's still interest here!

**Comment by mvisonneau** [comment]
Indeed, thanks for the guidance! FWIW, here is a snippet which seems to do the work:

```go
func isHelper(ctx *cli.Context) bool {
	for _, arg := range ctx.Args().Slice() {
		if arg == "--help" || arg == "-h" {
			return true
		}
	}
	return false
}

```
